Mathlib/Order/Interval/Finset/Fin.lean

```diff
@@ -166,6 +201,628 @@ theorem map_valEmbedding_Iic : (Iic b).map Fin.valEmbedding = Iic ↑b := by
 theorem map_valEmbedding_Iio : (Iio b).map Fin.valEmbedding = Iio ↑b := by
   rw [← attachFin_Iio, map_valEmbedding_attachFin]
 
+end val
+
+section castLE
+
+@[simp]
+theorem finsetImage_castLE_Iic (h : n ≤ m) :
+    (Iic a).image (castLE h) = Iic (castLE h a) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Iio (h : n ≤ m) :
+    (Iio a).image (castLE h) = Iio (castLE h a) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Icc (h : n ≤ m) :
+    (Icc a b).image (castLE h) = Icc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ioc (h : n ≤ m) :
+    (Ioc a b).image (castLE h) = Ioc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ico (h : n ≤ m) :
+    (Ico a b).image (castLE h) = Ico (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ioo (h : n ≤ m) :
+    (Ioo a b).image (castLE h) = Ioo (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_uIcc (h : n ≤ m) :
+    (uIcc a b).image (castLE h) = uIcc (castLE h a) (castLE h b) := by simp [← coe_inj]
```

Comment:
Could you please always follow the same order? I think `Icc Ico Ioc Ioo uIcc Ici Ioi Iic Iio` is fine for example